server/base/src/main/java/org/apache/accumulo/server/util/TabletServerLocks.java

```diff
@@ -41,38 +41,50 @@ static class Opts extends Help {
     String delete = null;
   }
 
-  public static void main(String[] args) throws Exception {
-
-    try (var context = new ServerContext(SiteConfiguration.auto())) {
-      String tserverPath = context.getZooKeeperRoot() + Constants.ZTSERVERS;
-      Opts opts = new Opts();
-      opts.parseArgs(TabletServerLocks.class.getName(), args);
+  public static void tabletServerLocks(String commandName, final ServerContext context,
+      final String lock, final boolean list, final String delete) throws Exception {
 
-      ZooCache cache = context.getZooCache();
-      ZooReaderWriter zoo = context.getZooReaderWriter();
+    String tserverPath = context.getZooKeeperRoot() + Constants.ZTSERVERS;
 
-      if (opts.list) {
+    ZooCache cache = context.getZooCache();
+    ZooReaderWriter zoo = context.getZooReaderWriter();
 
-        List<String> tabletServers = zoo.getChildren(tserverPath);
+    if (list) {
+      List<String> tabletServers = zoo.getChildren(tserverPath);
 
-        for (String tabletServer : tabletServers) {
-          var zLockPath = ServiceLock.path(tserverPath + "/" + tabletServer);
-          byte[] lockData = ServiceLock.getLockData(cache, zLockPath, null);
-          String holder = null;
-          if (lockData != null) {
-            holder = new String(lockData, UTF_8);
-          }
-
-          System.out.printf("%32s %16s%n", tabletServer, holder);
+      for (String tabletServer : tabletServers) {
+        var zLockPath = ServiceLock.path(tserverPath + "/" + tabletServer);
+        byte[] lockData = ServiceLock.getLockData(cache, zLockPath, null);
+        String holder = null;
+        if (lockData != null) {
+          holder = new String(lockData, UTF_8);
         }
-      } else if (opts.delete != null) {
-        ServiceLock.deleteLock(zoo, ServiceLock.path(tserverPath + "/" + args[1]));
-      } else {
-        System.out.println(
-            "Usage : " + TabletServerLocks.class.getName() + " -list|-delete <tserver lock>");
+
+        System.out.printf("%32s %16s%n", tabletServer, holder);
       }
+    } else if (delete != null) {
+      if (lock == null) {
+        printUsage(commandName);
+      }
+      ServiceLock.deleteLock(zoo, ServiceLock.path(tserverPath + "/" + lock));
+    } else {
+      printUsage(commandName);
+    }
+  }
+
+  public static void main(String[] args) throws Exception {
 
+    try (var context = new ServerContext(SiteConfiguration.auto())) {
```

Comment:
Should this command also aim to refactor the use of SiteConfiguration like in issue #2155?

Reply:
Yeah that is a good point. This one may need more refactoring to allow for that though. I think the main method should probably be dropped too if its going to be run through the Admin command. Then there should be a way to get the `ServerContext` from Admin.

Reply:
See Christopher's [comment](https://github.com/apache/accumulo/pull/2817#issuecomment-1208327394) about the difference between running in the Shell vs Admin.